batchflow/models/tf/encoder_decoder.py

```diff
@@ -328,26 +376,30 @@ def decoder(cls, inputs, name='decoder', **kwargs):
                     # Skip some of the steps
                     if factor[i] == 1:
                         continue
-                    # Upsample by a desired factor
-                    if upsample.get('layout') is not None:
-                        args = {**kwargs, **upsample, **unpack_args(upsample, i, steps)}
-                        x = cls.upsample(x, factor=factor[i], name='upsample-{}'.format(i), **args)
-
-                    # Post-process resulting tensor
-                    args = {**kwargs, **block_args, **unpack_args(block_args, i, steps)}  # enforce priority of subkeys
-                    x = base_block(x, name='post', **args)
 
-                    # Combine it with stored encoding of the ~same shape
+                    # Make all the args
+                    args = {**kwargs, **block_args, **unpack_args(block_args, i, steps)}
+                    upsample_args = {'factor': factor[i],
+                                     **kwargs, **upsample, **unpack_args(upsample, i, steps)}
+                    combine_args = {'op': args.get('combine_op'),
+                                    'data_format': args.get('data_format'),
+                                    **(skip if isinstance(skip, dict) else {})}
+
+                    if order == 'ub': # upsample -> block
+                        if upsample.get('layout') is not None:
+                            x = cls.upsample(x, name='upsample', **upsample_args)
+                        x = base_block(x, name='block-', **args)
+                    elif order == 'bu': # block -> upsample
+                        x = base_block(x, name='block', **args)
+                        if upsample.get('layout') is not None:
+                            x = cls.upsample(x, name='upsample', **upsample_args)
+                    else:
+                        raise ValueError('Unknown order {}, use one of ("ub", "bu")'.format(order))
+
+                    # Combine result with the stored encoding of the ~same shape
                     if skip and (i < len(inputs)-2):
```

Comment:
If skip in config is an empty dict, this check will not pass. Is this what you want?

Reply:
On one hand, `{}` evaluates to `False` and should be a sign of that.
On the other hand, in this situation it can also mean 'I want to combine tensors with no additional parameters'.